lms/djangoapps/grades/management/commands/compute_grades.py

```diff
@@ -0,0 +1,122 @@
+"""
+Command to compute all grades for specified courses.
+"""
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import logging
+
+from django.core.management.base import BaseCommand, CommandError
+import six
+
+from openedx.core.lib.command_utils import (
+    get_mutually_exclusive_required_option,
+    parse_course_keys,
+)
+from student.models import CourseEnrollment
+from xmodule.modulestore.django import modulestore
+
+from ... import tasks
+
+
+log = logging.getLogger(__name__)
+
+
+class Command(BaseCommand):
+    """
+    Example usage:
+        $ ./manage.py lms compute_grades --all_courses --settings=devstack
+        $ ./manage.py lms compute_grades 'edX/DemoX/Demo_Course' --settings=devstack
+    """
+    args = '<course_id course_id ...>'
+    help = 'Computes grade values for all learners in specified courses.'
+
+    def add_arguments(self, parser):
+        """
+        Entry point for subclassed commands to add custom arguments.
+        """
+        parser.add_argument(
+            '--courses',
+            dest='courses',
+            nargs='+',
+            help='List of (space separated) courses that need grades computed.',
+        )
+        parser.add_argument(
+            '--all_courses',
+            help='Compute grades for all courses.',
+            action='store_true',
+            default=False,
+        )
+        parser.add_argument(
+            '--routing_key',
+            dest='routing_key',
+            help='Celery routing key to use.',
+        )
+        parser.add_argument(
+            '--batch_size',
+            help='Maximum number of students to calculate grades for, per celery task.',
+            default=100,
+            type=int,
+        )
+        parser.add_argument(
+            '--start_index',
+            help='Offset from which to start processing enrollments.',
+            default=0,
+            type=int,
+        )
+
+    def handle(self, *args, **options):
+        self._set_log_level(options)
+
+        for course_key in self._get_course_keys(options):
+            self.enqueue_compute_grades_for_course_tasks(course_key, options)
+
+    def enqueue_compute_grades_for_course_tasks(self, course_key, options):
+        """
+        Enqueue celery tasks to compute and persist all grades for the
+        specified course, in batches.
+        """
+        enrollment_count = CourseEnrollment.objects.num_enrolled_in(course_key)
```

Comment:
This checks only for `active` learners.  I assume we want to backfill grades for all engaged learners, regardless of whether they have un-enrolled from the course.  In fact, analyzing engaged learners who eventually un-enrolled may provide interesting information.